lib/task.ml

```diff
@@ -79,15 +140,16 @@ let parallel_for_reduce ?(chunk_size=0) ~start ~finish ~body pool reduce_fun ini
       loop (s+1) (body s)
     else begin
       let d = s + ((e - s) / 2) in
-      let p = async pool (fun _ -> work s d) in
+      let p = async ~pool (fun _ -> work s d) in
       let right = work (d+1) e in
-      let left = await pool p in
+      let left = await ~pool p in
       reduce_fun left right
     end
   in
   reduce_fun init (work start finish)
 
-let parallel_for ?(chunk_size=0) ~start ~finish ~body pool =
+let parallel_for ?pool ?(chunk_size=0) ~start ~finish ~body () =
```

Comment:
I'm not happy with introducing a unit parameter in the `parallel_for` function. This had to be done since at least one argument needs to be non-labelled. Alternatively we can remove the label for `body` argument.